my-sms/app/views/courses/index.html.haml

```diff
@@ -0,0 +1,3 @@
+- display_attrs = %i[id name number_of_semesters allocation start_date end_date]
+= render partial: 'shared/crud_models',
+  locals: {model_class: Course, model_collection: @courses, display_attrs: display_attrs}
```

Comment:
I like the idea, although you may think about using Presenters/Decorators (very similar patterns) to make your shared views more manageable and secure. For example, `display_attrs` could be extracted to Presenters. Check out this pattern: https://gist.github.com/somebox/5a7ebf56e3236372eec4#in-the-view

Reply:
That's interesting. Although I would prefer to continue using the partials pattern. 

The idea stated in the gist is to move business logic from controller and views. However, here the logic is purely view logic, which is traditionally separated into the View component. This also matches the `form_for` pattern logically (but not syntactically), in terms of what's inherited. 

I'm conscious of a rails trend to minimise code in views even if that code is view logic. Similar to desktop MVC where the view can directly observe the model and the 'view' part of the separation is preferred to act as templates rather than separating all view functionality. In that case, as you mention, I could move some logic into decorators that are already set up (such as `display_attrs`) but imo the separation is better here because the granularity is at the view level. For example, `display_attrs` would make also an appearance when creating the course edit/new pages but with different attrs.